mobile/lib/services/location_service.dart

```diff
@@ -187,6 +158,18 @@ class LocationService {
       );
     }
 
+    final permissionGranted =
+        await PermissionService.checkPermission(AppPermission.location);
+
+    if (!permissionGranted) {
+      return;
+    }
+
+    bool? serviceEnabled = await Geolocator.isLocationServiceEnabled();
+    if (!serviceEnabled) {
+      return;
+    }
```

Comment:
Checking if a user has permitted the app to access current location and has turned on location